update-outdoor_temp.ts

```diff
@@ -0,0 +1,133 @@
+import axios from "axios";
+import { Pool } from "pg";
+
+interface WeatherData {
+    hourly: {
+        time: string[];
+        temperature_2m: number[];
+    };
+}
+
+if (!process.env.PG_CONNECTION_STRING) {
+    console.error(
+        "The PG_CONNECTION_STRING environment variable must be defined.",
+    );
+}
+
+const pool = new Pool({
+    connectionString: process.env.PG_CONNECTION_STRING,
+});
+
+const client = await pool.connect();
+
+try {
+    const res = await client.query(
+        "SELECT id, start_latlng, start_date, elapsed_time FROM activities WHERE outdoor_temp IS NULL",
+    );
+    const activities = res.rows;
+
+    for (const activity of activities) {
+        const { id, start_latlng, start_date, elapsed_time } = activity;
+
+        if (start_latlng && start_latlng.length === 2) {
+            const [latitude, longitude] = start_latlng;
+            let weatherData: WeatherData | null = null;
+
+            const startDate = new Date(start_date);
+            const endDate = new Date(startDate.getTime() + elapsed_time * 1000);
+            const daysAgo =
+                (new Date().getTime() - startDate.getTime()) /
+                (1000 * 60 * 60 * 24);
+
+            let apiEndpoint: string;
+            let params: any;
+
+            if (daysAgo > 3) {
+                apiEndpoint = "https://archive-api.open-meteo.com/v1/era5";
```

Comment:
So is this like a free API, no need for authentication of any sort?

Reply:
Yup - it's free for non-commercial use. They don't restrict access, but ask for fair use and attribution. More info [here](https://open-meteo.com/en/pricing)